web/src/dropdown_list_widget.js

```diff
@@ -28,7 +29,9 @@ export class DropdownListWidget {
         this.null_value = null_value;
         this.include_current_item = include_current_item;
         this.initial_value = value;
+        this.render_pre_dropdown_icon = render_pre_dropdown_icon;
         this.on_update = on_update;
+        this.list_widget = null;
```

Comment:
Can these changes be a new series of commits, either independently or at the start of the PR? I think it should be easy to extract with `git checkout -p` and an interactive rebase... but I think there's 2 new features here -- the `replace_data` method and `render_pre_dropdown_icon`?

Reply:
Yeah, that makes sense as a prep commit, thanks! I'm building a model for when to consider something to be a separate commit -- changes to general use modules is one I'll keep an eye on going forward.

Reply:
Split into its own commit at the start of the PR :) Let me know if you want it in a different PR